UserRelated/Main.py

```diff
@@ -0,0 +1,208 @@
+    and initializes it with all of the relevant arguments.
+    :return: a command-line arguments parser.
+    """
+    parser = argparse.ArgumentParser(description='Gets arguments from the user.')
+    parser.add_argument("search_area", help="The general geographic area to search tracks in.",
+                        choices=['baiersbronn'])  # Other areas options in the future :)
+
+    add_limits_args(parser)
+    add_interest_points_args(parser)
+
+    parser.add_argument("length", help="1 for a short track, 2 for medium-length and 3 for long", type=int)
+    parser.add_argument("difficulty", help="1 for an easy track, 2 for intermediate and 3 for difficult", type=int)
+    parser.add_argument("shape", help="1 for a loop and 2 for out and back", type=int)
+
+    return parser
+
+
+def add_interest_points_shingles(shing_set: set, args: argparse.Namespace):
+    """
+    Adds interest points shingles shingles to shing_set.
+    :param shing_set: the user preferences shingle set.
+    :param args: the given command-line arguments.
+    """
+    if args.waterfall: shing_set.add(PointTag.WATERFALL.value)
+    if args.birding: shing_set.add(PointTag.BIRDING.value)
+    if args.river: shing_set.add(PointTag.RIVER.value)
+    if args.cave: shing_set.add(PointTag.CAVE.value)
+    if args.lake: shing_set.add(PointTag.WATER.value)
+    if args.spring: shing_set.add(PointTag.SPRING.value)
+    if args.geo: shing_set.add(PointTag.GEOLOGIC.value)
+    if args.historic: shing_set.add(PointTag.HISTORIC.value)
+
+
+def add_length_shingle(shing_set: set, args: argparse.Namespace):
+    """
+    Adds the appropriate length shingle to the shingles set.
+    :param shing_set: the user preferences shingle set
+    :param args: the given command-line arguments.
+    """
+    if args.length == 1:
+        shing_set.add(TrackLength.SHORT.value)
+    elif args.length == 2:
+        shing_set.add(TrackLength.MEDIUM.value)
+    else:
+        shing_set.add(TrackLength.LONG.value)
+
+
+def add_difficulty_shingle(shing_set: set, args: argparse.Namespace):
+    """
+    Adds the appropriate difficulty shingle to the shingles set.
+    :param shing_set: the user preferences shingle set
+    :param args: the given command-line arguments.
+    """
+    if args.difficulty == 1:
+        shing_set.add(TrackDifficulty.EASY.value)
+    elif args.difficulty == 2:
+        shing_set.add(TrackDifficulty.INTERMEDIATE.value)
+    else:
+        shing_set.add(TrackDifficulty.DIFFICULT.value)
+
+
+def add_shape_shing(shing_set: set, args: argparse.Namespace):
+    """
+    Adds the appropriate shape shingle to the shingles set.
+    :param shing_set: the user preferences shingle set
+    :param args: the given command-line arguments.
+    """
+    shing_set.add(TrackShape.LOOP.value) if args.shape == 1 else shing_set.add(TrackShape.CURVE.value)
+
+
+# Todo: this is a code duplication of the slopes_poc branch, we should have a separate class for LSH-ing we can reuse.
```

Comment:
i didn't understand: i think this should be here..
are you talking about the pass over every shin in shingles?
i think this func should be creating the shingles, while questioning SlopeMap. 

 we can discuss the design if you like:)

Reply:
What I meant is: since we use LSH to find similarity in two different situations (difficulty evaluation, finding a match for the user's request) we might want to think of a general module responsible for similarity finding to avoid code duplication. I think that other for finding shingles for the compared objects, the rest of the similarity finding process is similar no matter what we're comparing or for what use. 
I think it's a topic for future improvement and shouldn't interfere with the merge, I added it to the project backlog.